metadata-ingestion/src/datahub/ingestion/source/kafka.py

```diff
@@ -145,6 +152,49 @@ def report_dropped(self, topic: str) -> None:
         self.filtered.append(topic)
 
 
+class KafkaConnectionTest:
+    def __init__(self, config_dict: dict):
+        self.config = KafkaSourceConfig.parse_obj_allow_extras(config_dict)
+        self.report = KafkaSourceReport()
+        self.consumer: confluent_kafka.Consumer = confluent_kafka.Consumer(
```

Comment:
this logic to construct a confluent_kafka.Consumer is duplicated - can we extract it out into a helper method?